test/integrationtests/voight_kampff/tools.py

```diff
@@ -15,15 +15,161 @@
 """Common tools to use when creating step files for behave tests."""
 
 from threading import Event
+from typing import Any, Callable, List, Tuple
 import time
 
 from mycroft.audio.utils import wait_while_speaking
 from mycroft.messagebus import Message
 
-
 DEFAULT_TIMEOUT = 10
 
 
+class VoightKampffEventMatcher:
+    """Matches a specified message type to messages emitted on the bus.
+
+    Attributes:
+        message_type: identifier of the message to search for on the bus
+        context: the Behave context from the test utilizing this class
+        match_event: mechanism for knowing when a match is found
+        error_message: message that can be used by the test to communicate
+            the reason for a failed match to the tester.
+    """
+    def __init__(self, message_type: str, context: Any):
+        self.message_type = message_type
+        self.context = context
+        self.match_event = Event()
+        self.match_found = False
+        self.error_message = ""
+
+    def match(self, timeout: int = None):
+        """Attempts to match the requested message type to emitted bus events.
+
+        Use a message bus event handler to capture any message emitted on the
+        bus that matches the message type specified by the caller.  Also
+        checks any messages emitted prior to the handler being defined to
+        protect against a race condition.
+
+        Args:
+            timeout: number of seconds to attempt matching before giving up
+        """
+        timeout = timeout or self.context.step_timeout
+        self.context.bus.on(self.message_type, self.handle_message)
+        self._check_historical_messages()
+        if not self.match_event.is_set():
+            self.match_event.wait(timeout=timeout)
+        self.context.bus.remove(self.message_type, self.handle_message)
+        self.match_found = self.match_event.is_set()
+        if not self.match_found:
+            self._build_error_message()
```

Comment:
I think the match should return the success / failure. Passing this information through state seems contra-intuitive.

Reply:
Could you please elaborate?  I am not understanding what is iffy about using the instance attributes.

Reply:
I think the last change I just pushed to use wrapper functions alleviates this concern.